src/zcash/NoteEncryption.cpp

```diff
@@ -48,6 +101,155 @@ void KDF(unsigned char K[NOTEENCRYPTION_CIPHER_KEYSIZE],
 
 namespace libzcash {
 
+boost::optional<SaplingNoteEncryption> SaplingNoteEncryption::FromDiversifier(diversifier_t d) {
+    uint256 epk;
+    uint256 esk;
+
+    // Pick random esk
+    librustzcash_sapling_generate_r(esk.begin());
```

Comment:
Note that the beta-21 spec says that esk should not be 0. (This is a consequence of the implementation already checking that epk is not of small order.) This has negligible probability so it's not so important.